uni/lib/view/home/widgets/bus_stop_card.dart

```diff
@@ -44,132 +98,62 @@ class BusStopCard extends GenericCard {
   void onRefresh(BuildContext context) {
     Provider.of<BusStopProvider>(context, listen: false).forceRefresh(context);
   }
-}
 
-/// Returns a widget with the bus stop card final content
-Widget getCardContent(
-  BuildContext context,
-  Map<String, BusStopData> stopData,
-  RequestStatus busStopStatus,
-) {
-  switch (busStopStatus) {
-    case RequestStatus.successful:
-      if (stopData.isNotEmpty) {
-        return Column(
-          children: <Widget>[
-            getCardTitle(context),
-            getBusStopsInfo(context, stopData)
-          ],
-        );
-      } else {
-        return Container(
-          padding: const EdgeInsets.all(8),
-          child: Row(
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: <Widget>[
-              Text(
-                S.of(context).buses_personalize,
-                maxLines: 1,
-                overflow: TextOverflow.ellipsis,
-                style: Theme.of(context).textTheme.titleSmall!.apply(),
-              ),
-              IconButton(
-                icon: const Icon(Icons.settings),
-                onPressed: () => Navigator.push(
-                  context,
-                  MaterialPageRoute<BusStopSelectionPage>(
-                    builder: (context) => const BusStopSelectionPage(),
-                  ),
-                ),
-              )
-            ],
+  /// Returns a list of widgets for each bus stop info that exists
+  List<Widget> getEachBusStopInfo(
+    BuildContext context,
+    Map<String, BusStopData> stopData,
+  ) {
+    final rows = <Widget>[const LastUpdateTimeStamp<BusStopProvider>()];
+
+    stopData.forEach((stopCode, stopInfo) {
+      if (stopInfo.trips.isNotEmpty && stopInfo.favorited) {
```

Comment:
Just to make this clearer: Can we do `stopData.where(...).map(...)` instead of this imperative approach?